tools/wpt/browser.py

```diff
@@ -105,36 +107,30 @@ def platform_string_geckodriver(self):
 
         return "%s%s" % (platform, bits)
 
-    def latest_nightly_listing(self):
-        resp = get("https://archive.mozilla.org/pub/firefox/nightly/latest-mozilla-central/")
-        resp.raise_for_status()
-        return resp.text
-
-    def get_nightly_link(self, index, platform):
-        pattern = re.compile("<a[^>]*>(firefox-(\d+)\.\d(?:\w\d)?.en-US.%s\.tar\.bz2)" % platform)
-        max_version = None
-        for match in pattern.finditer(index):
-            try:
-                version = int(match.group(2))
-            except ValueError:
-                continue
-            if max_version is None or version > max_version[0]:
-                max_version = (version, match.group(1))
-        if not max_version:
-            raise ValueError("Failed to find version to download")
-        return ("https://archive.mozilla.org/pub/firefox/nightly/latest-mozilla-central/%s" %
-                max_version[1])
-
     def install(self, dest=None):
         """Install Firefox."""
+
         if dest is None:
             dest = os.getcwd()
 
-        nightly_link = self.get_nightly_link(self.latest_nightly_listing(),
-                                             self.platform_string())
-        resp = get(nightly_link)
-        resp.raise_for_status()
-        untar(resp.raw, dest=dest)
+        filename = FactoryScraper('daily', branch='mozilla-central', destination=dest).download()
+
+        try:
+            mozinstall.install(filename, dest)
+        except mozinstall.mozinstall.InstallError as e:
+            platform = {
+                "Linux": "linux",
+                "Windows": "win",
+                "Darwin": "mac"
+            }.get(uname[0])
+
+            if platform == "mac":
+                mozinstall.uninstall(dest+'/Firefox Nightly.app')
+                mozinstall.install(filename, dest)
+            else:
+                raise
+
+        os.remove(filename)
         return find_executable("firefox", os.path.join(dest, "firefox"))
```

Comment:
Should we consider `mozinstall.get_binary` here?

Reply:
We could, but I have no way of testing this on a Mac anymore.